### PR TITLE
Add support for TaxRate

### PR DESCRIPTION
**Pull request by remi-stripe**
This PR adds support for the `TaxRate` resource and APIs. It also modifies all relevant objects that are impacted by this new ship.

cc @stripe/api-libraries 
cc @vihari-stripe @george-stripe

### CODE REVIEW

**Comment by remi-stripe** [comment]
This is now ready for review. r? @mickjermsurawong-stripe

**Comment by remi-stripe** [comment]
re-assigning to @brandur-stripe as @mickjermsurawong-stripe is working on finishing stripe-java

File: taxrate.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	Created      int64             `json:"created"`
+	Description  string            `json:"description"`
+	DisplayName  string            `json:"display_name"`
+	ID           string            `json:"id"`
+	Inclusive    bool              `json:"inclusive"`
+	Jurisdiction string            `json:"jurisdiction"`
+	Livemode     bool              `json:"livemode"`
+	Metadata     map[string]string `json:"metadata"`
+	Object       string            `json:"object"`
+	Percentage   float64           `json:"percent_off"`
```
Comment (brandur-stripe) [inline taxrate.go]: I think this one should be `json:"percentage"`.

Comment (remi-stripe) [reply taxrate.go]: nice catch

**Comment by remi-stripe** [comment]
Going to self-approve after fixing the issue! Will merge once tests pass

**Pull request merged**